pkg/smbios/smbios_modifier.go

```diff
@@ -42,28 +41,96 @@ func getEntries(smbiosBase func() (int64, int64, error), memFile *os.File) (*Ent
 	return e32, e64, entryAddr, err
 }
 
-// OverrideOpt is a function overriding the marshaler
-type OverrideOpt func(over map[TableType]encoding.BinaryMarshaler)
-
-// ReplaceTable returns func replacing the marshaler given table type
-func ReplaceTable(typ TableType, table encoding.BinaryMarshaler) OverrideOpt {
-	return func(over map[TableType]encoding.BinaryMarshaler) {
-		over[typ] = table
+// OverrideOpt is a function return overridden tables given another tables the marshaler
+type OverrideOpt func(t []*Table) ([]*Table, error)
+
+// ReplaceSystemInfo returns override options of the SystemInfo table with the given values
+func ReplaceSystemInfo(manufacturer, productName, version, serialNumber, sku, family *string, uuid *UUID, wakeupType *WakeupType) OverrideOpt {
+	return func(tables []*Table) ([]*Table, error) {
+		var result []*Table
+
+		for _, t := range tables {
+			if t.Type != TableTypeSystemInfo {
+				result = append(result, t)
+				continue
+			}
+			// replace it
+			si, err := ParseSystemInfo(t)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse system info: %w", err)
+			}
+
+			if manufacturer != nil {
+				si.Manufacturer = *manufacturer
+			}
+			if productName != nil {
+				si.ProductName = *productName
+			}
+			if version != nil {
+				si.Version = *version
+			}
+			if serialNumber != nil {
+				si.SerialNumber = *serialNumber
+			}
+			if sku != nil {
+				si.SKUNumber = *sku
+			}
+			if family != nil {
+				si.Family = *family
+			}
+			if uuid != nil {
+				si.UUID = *uuid
+			}
+			if wakeupType != nil {
+				si.WakeupType = *wakeupType
+			}
+
+			sit, err := si.toTable()
+			if err != nil {
+				return nil, fmt.Errorf("failed to convert system info to table: %w", err)
+			}
+			result = append(result, sit)
+		}
+		return result, nil
 	}
 }
 
-// ModifySystemInfo modifies the SystemInfo table in system memory
-func (m *Modifier) ModifySystemInfo(manufacturer, productName, version, serialNumber string) error {
-	sysInfo, err := m.Info.GetSystemInfo()
-	if err != nil {
-		return fmt.Errorf("failed to get system info: %w", err)
+// ReplaceBaseboardInfoMotherboard returns override options that only overrides table with Type = BaseboardInfo and BoardType = BoardTypeMotherboardIncludesProcessorMemoryAndIO
+func ReplaceBaseboardInfoMotherboard(assetTag *string) OverrideOpt {
```

Comment:
I think this is ok for now, but there's gotta be a way to have a better API. I'm good to merge this one though and improve it in the other repo.

Reply:
Noted, thanks.